go/saltpack/decrypt.go

```diff
@@ -141,6 +144,7 @@ func (ds *decryptStream) tryVisibleReceivers(hdr *EncryptionHeader, ephemeralKey
 			kids = append(kids, r.ReceiverKID)
```

Comment:
Not part of this diff, but I'm worried that this loop does the wrong thing if any anonymous receivers come before any public receivers in `hdr.Receivers`. `LookupBoxSecretKey` will return the index in the `kids` list, right, not in the actual `hrd.Receivers` list?